### ISSUE TITLE
Sourcehut / sr.ht backend

### ISSUE THREAD

**Issue by fgaz** [haskell-CI/haskell-ci#526]
https://sourcehut.org/ 's CI has some nice features, and it'd be nice to support it.

I have a basic prototype here: https://github.com/fgaz/haskell-ci/tree/sourcehut

It's very rough and barebones for now, but it does work. I'm not sure when/if I'll be able to polish this up, but I thought I'd open this ticket in the meantime so if someone else is interested there's no duplicate work.

**Comment by phadej** [comment]
That project is not created by me. I'll remove that one.

**Comment by fgaz** [comment]
@phadej i polished that branch a bit and it's now almost in a mergeable state. In particular I'm now using ListBuilder like in the github one, so it should be easy to extend it.

Would you accept a pr even if it only has a few basic steps for now?

**Comment by fgaz** [comment]
Makes sense. I probably won't have time to make it reach or keep feature parity with github either, but something is better than nothing. Yes, I'll try to fix any backend-specific issues that arise.

I'll open a pr then.
